scipy/optimize/_trustregion_iterative.py

```diff
@@ -0,0 +1,470 @@
+"""Iterative trust-region optimization subproblem."""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.linalg import norm, get_lapack_funcs, solve_triangular, norm
+from ._trustregion import (_minimize_trust_region, BaseQuadraticSubproblem)
+
+__all__ = ['estimate_smallest_singular_value',
+           'singular_leading_submatrix',
+           'IterativeSubproblem']
+
+
+def _minimize_trust_iterative(fun, x0, args=(), jac=None, hess=None,
+                              **trust_region_options):
+    """
+    Minimization of scalar function of one or more variables using
+    a iterative trust-region algorithm.
+
+    Options
+    -------
+    initial_trust_radius : float
+        Initial trust-region radius.
+    max_trust_radius : float
+        Maximum value of the trust-region radius. No steps that are longer
+        than this value will be proposed.
+    eta : float
+        Trust region related acceptance stringency for proposed steps.
+    gtol : float
+        Gradient norm must be less than ``gtol` before successful
+        termination.
+    trust_region_factor, optimization_factor : float
+        Parameters that determines the stop criteria for the subproblem solver.
+        The iterative algoritm for the trust-region subproblem will stop when
+        the computed step ``p`` fullfil
+        ``norm(p)<=trust_region_factor*trust_radius``
+        or when the solution ``m`` for the subproblem fullfil
+        ``m < optimization_factor*m_optimal``, with ``m_optimal`` being the
```

Comment:
This condition does make sense if you think about it. Supose you have a subproblem that have a local minima and you intend to go in the descent direction towards that local minima. For a step size ``trust_radius`` the value of your cost function is ``m_optimal`` if, however, you allow your step size to be a greater than  ``trust_radius``  since you are going into a descent direction that means that ``m<m_optimal``. Note, however, that this violation is not good, because you are violating your step size.

But you are right, this is not a very intuitive parameter. By the tests I have been doing I realized that the default values of both ``trust_radius_factor`` and ``optimization_factor`` provide pretty good results. That way, I think I can remove then from the public interface without great loss. 

I intend, however, to keep then in the constructor of the class ``IteractiveSubproblem`` (which is not visible for the users). I have three reasons for doing that: i) it allows others developers to change it easily in the future; ii) I can describe those parameters on the class doc, which I think is nice. iii) I can use these parameters in order to test if the subproblem solver is working well (as I am doing now).
Are you ok with that?